### PR TITLE
User fallback oauth

### PR DESCRIPTION
**Pull request by pfych**
When a user interacts with the bot ensure they are Authed

### CODE REVIEW

File: src/server/server.ts [hunk 1]
```diff
@@ -87,12 +87,18 @@ app.get("/oauth/callback", async (req, res) => {
...
+	const tokenRes = await TachiServerV1Request<APITokenDocument>(
+		RequestTypes.POST,
+		"/oauth/token",
+		"2e94ea15768d93d6621e60e58c94d351464ab71f",
+		{
+			code: req.query.code,
+			client_id: ProcessEnv.BOT_CLIENT_ID,
+			client_secret: ProcessEnv.BOT_CLIENT_SECRET,
+			grant_type: "authorization_code",
+			redirect_uri: "http://localhost:3001/oauth/callback"
```
Comment (pfych) [inline src/server/server.ts:99]: This will need to change :skull:

File: src/server/server.ts [hunk 2]
```diff
@@ -87,12 +87,18 @@ app.get("/oauth/callback", async (req, res) => {
...
-	const tokenRes = await TachiServerV1Request<APITokenDocument>(RequestTypes.POST, "/oauth/token", null, {
-		code: req.query.code,
-		client_id: ProcessEnv.BOT_CLIENT_ID,
-		client_secret: ProcessEnv.BOT_CLIENT_SECRET,
-		redirect_uri: `${BotConfig.OUR_URL}/oauth/callback`
-	});
+	const tokenRes = await TachiServerV1Request<APITokenDocument>(
+		RequestTypes.POST,
+		"/oauth/token",
+		"2e94ea15768d93d6621e60e58c94d351464ab71f",
```
Comment (pfych) [inline src/server/server.ts:93]: Lmao :skull: 

Invalidated

File: src/server/server.ts [hunk 3]
```diff
@@ -123,10 +129,9 @@ app.get("/oauth/callback", async (req, res) => {
 		userID: user.id
 	});
 
-	return res.sendFile("../pages/account-linked.html");
+	return res.sendFile(__dirname + "/../pages/account-linked.html");
```
Comment (pfych) [inline src/server/server.ts:132]: This never worked and this "fix" doesn't work either. Revert.

File: src/server/server.ts [hunk 4]
```diff
@@ -90,13 +90,13 @@ app.get("/oauth/callback", async (req, res) => {
 	const tokenRes = await TachiServerV1Request<APITokenDocument>(
 		RequestTypes.POST,
 		"/oauth/token",
-		"2e94ea15768d93d6621e60e58c94d351464ab71f",
+		"GIVE_BOT_A_VALID_TOKEN_TO_HIT_AUTH",
```
Comment (pfych) [inline src/server/server.ts:93]: @zkldi What token should the bot use to create tokens?

Would it be possible to make this endpoint unauthenticated?

**Comment by zkldi** [comment]
This has been merged, but I had to resolve some conflicts locally and i accidentally did it against main instead of against here.
regardless, we should be good.

**Pull request closed without merging**